core/src/main/java/org/testcontainers/dockerclient/EnvironmentAndSystemPropertyClientProviderStrategy.java

```diff
@@ -15,11 +18,25 @@
     public static final int PRIORITY = 100;
 
     // Try using environment variables
-    private final DockerClientConfig dockerClientConfig = DefaultDockerClientConfig.createDefaultConfigBuilder().build();
+    private final DockerClientConfig dockerClientConfig;
+
+    public EnvironmentAndSystemPropertyClientProviderStrategy() {
+        DefaultDockerClientConfig.Builder configBuilder = DefaultDockerClientConfig.createDefaultConfigBuilder();
+
+        getUserProperty("docker.host").ifPresent(configBuilder::withDockerHost);
```

Comment:
Question - what wins, environment variable or a user property?

Reply:
`getEnvVarOrUserProperty` always looks at env vars prefixed with `TESTCONTAINERS_`, so the order of resolution would be:

* `TESTCONTAINERS_DOCKER_HOST` env var
* `docker.host` in `~/.testcontainers.properties`
* Fall back to docker-java's own resolution, which I believe is:
	* ~`DOCKER_HOST` system property~
	* `DOCKER_HOST` env var
	* ~`DOCKER_HOST` in `~/.docker-java.properties`~

The struck out items are those which docker-java is _capable_ of resolving but which wouldn't trigger the `isApplicable()` method, therefore would not be effective.

Note that `TESTCONTAINERS_DOCKER_HOST` seems unlikely to be useful, but I think we should avoid adding yet another means of obtaining settings in addition to the `getEnvVar`, `getEnvVarOrUserProperty`, `getEnvVarOrProperty` hierarchy that's used for everything* else.

* there are still some `getEnv("DOCKER_HOST")`s floating around, e.g. in the Compose and I think Localstack support which we should tidy up in another PR.

Reply:
> Note that TESTCONTAINERS_DOCKER_HOST seems unlikely to be useful, but I think we should avoid adding yet another means of obtaining settings in addition to the getEnvVar, getEnvVarOrUserProperty, getEnvVarOrProperty hierarchy that's used for everything* else.

What if we add "if name starts with `docker.` then avoid adding `TESTCONTAINERS_` prefix"?

> there are still some getEnv("DOCKER_HOST")s floating around

really? Wow, this is bad. They really should be reading from the selected strategy.

Reply:
> What if we add "if name starts with docker. then avoid adding TESTCONTAINERS_ prefix"?

If we do this, then `Also, I think DOCKER_HOST environment variable must override docker.host in ~/.testcontainers.properties` is achieved, so 🤷‍♂️ OK!